src/components/idle-picker/IdlePicker.js

```diff
@@ -0,0 +1,193 @@
+            case 'LEFT':
+            case 'ArrowLeft':
+            case 'DOWN':
+            case 'ArrowDown':
+                pickerValue > 1 && this.updateValue(pickerValue - 1, pickerValue - 1, true, true);
+                break;
+            case 'RIGHT':
+            case 'ArrowRight':
+            case 'UP':
+            case 'ArrowUp':
+                pickerValue < 10 && this.updateValue(pickerValue + 1, pickerValue + 1, true, true);
+                break;
+            default:
+            }
+        };
+
+        handle.addEventListener('mouseenter', appendClass);
+        handle.addEventListener('mouseleave', removeClass);
+        handle.addEventListener('focusin', appendClass);
+        handle.addEventListener('blur', removeClass);
+
+        handle.addEventListener('keydown', keyboardMove);
+
+        handle.addEventListener('mousedown', setClick);
+        handle.addEventListener('mouseup', () => unsetClick(true));
+        window.addEventListener('mouseup', () => unsetClick(false));
+    }
+
+    render() {
+        const { pickerValue, displayValue, forcePickerValue } = this.state;
+
+        const pickerProps = {
+            min: 1,
+            max: 10,
+            step: 0.01,
+            marks,
+            onChange: this.update,
+            onAfterChange: this.afterChange,
+            onBeforeChange: this.beforeChange,
+        };
+        const finalClassName = classNames(
+            styles['picker-container']
+        );
+
+        if (forcePickerValue === true) {
+            pickerProps.value = pickerValue;
+            pickerProps.className = classNames(styles['rc-slider-main'], styles['slider-component']);
+        } else {
+            pickerProps.className = classNames(styles['is-focused'], styles['slider-component']);
+        }
+
+        return (
+            <div className={ finalClassName } ref={ this.componentRef }>
+                <Display currentValue={ displayValue } />
+                <RcSlider { ...pickerProps } />
+            </div>
+        );
+    }
+
+    beforeChange = () => {
+        this.previousHandleWasBeforeChange = true;
+        this.updateValue(null, null, false);
+    };
+
+    update = (value) => {
+        const roundValue = Math.round(value);
+
+        if (roundValue !== this.state.displayValue) {
+            if (this.previousHandleWasBeforeChange) {
+                this.updateValue(roundValue, roundValue, true);
+            } else {
+                this.updateValue(null, roundValue, false);
+            }
+        }
+        this.previousHandleWasBeforeChange = false;
+    };
+
+    afterChange = (value) => {
+        this.previousHandleWasBeforeChange = false;
+        const roundedValue = Math.round(value);
+
+        this.updateValue(roundedValue, roundedValue, true, true);
+        this.props.handlePickerChange && this.props.handlePickerChange(roundedValue);
+    };
+
+    updateValue = (pickerValue, displayValue, forcePickerValue) => {
```

Comment:
Instead of having this method, couldn't we simply use `this.setState` in the places it's being called? Seems simpler to me.

Reply:
Agreed. This method was extracted because there was other work being done there in a previous approach, so it no longer makes as much sense to have this method.